### ISSUE TITLE
Feature Request: TypeScript migration

### ISSUE THREAD

**Issue by kevinweejh** [kevinweejh/resu-me#12]
**Is your feature request related to a problem? Please describe.**
Not necessarily a problem per se. This project was created before I understood the long-term benefits of maintaining a type safe TS project over a JS one. This feature request is being raised to migrate the entire project to use TS before further development. 

**Describe the solution you'd like**
Install and implement TypeScript in this existing JavaScript Vite project. This is part 1 of 2. 
Part 1 involves the set-up, installing of necessary packages, adjusting config files and entry file. 
Part 2 involves the migration of components from JSX to TSX. 

**Describe alternatives you've considered**
NIL 

**Additional context**
This is a pre-emptive feature that, ideally, should be done before further development - so as to keep the migration as simple as possible. 

**Possible Implementation**
Refer to [this post](https://github.com/vitejs/vite/discussions/6799#discussioncomment-5393727) for guidance.

**Benefits**
Reduce bugs that make it to production, improve code quality, expand IDE support, and enhance developer productivity

**Risks**
NIL

**Comment by kevinweejh** [comment]
Part 1 of 2 complete - all configurations and entry file updated. 
New issue will be created for Part 2 of 2, which is for migration of all components.
